src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -768,20 +768,20 @@ internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol
             TypeSymbol targetType,
             BindingDiagnosticBag diagnostics)
         {
-            var collectionTypeKind = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, targetType, out var elementType);
+            var collectionTypeKind = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, targetType, out TypeWithAnnotations elementTypeWithAnnotations);
```

Comment:
We may want to have you update your PR branch after some of the optimization PRs get merged. The change from Type to TypeWithAnnotations likely will conflict